UI/src/model/LocationSearch.js

```diff
@@ -14,30 +22,176 @@
       });
     };
 
-    var getCoordinatesFromRoadAddress = function(road) {
+    /**
+     * Combined numerical value search (asset id and road number, which is part of road address)
+     *
+     * @param input
+     * @returns {*}
+     */
+    var idOrRoadNumber = function(input) {
+      if (selectedLayer === 'massTransitStop') {
+        return roadNumberAndNationalIdSearch(input);
+      } else if (selectedLayer === 'linkProperty') {
+        return roadNumberAndRoadLinkSearch(input);
+      } else if (selectedLayer === 'speedLimit') {
+        return roadNumberAndSpeedLimitSearch(input.text);
+      } else {
+        return getCoordinatesFromRoadAddress({roadNumber: input.text});
+      }
+    };
+
+    /**
+     * Speed limit id search, combined with road number search
+     *
+     * @param input
+     * @returns {*}
+     */
+    var roadNumberAndSpeedLimitSearch = function(input){
+      var roadNumberSearch = backend.getCoordinatesFromRoadAddress(input);
+      var speedlimitSearch= backend.getSpeedLimitsLinkIDFromSegmentID(input);
+      return $.when(
+        speedlimitSearch, roadNumberSearch).then(function(speedlimitdata,roadData) {
+        var returnObject = roadLocationAPIResultParser(roadData);
+        var linkFound =_.get(speedlimitdata[0], 'success');
+        if (linkFound === true) {
```

Comment:
In this case I thing that you can remove the variable 'link Found' and put directly the content in the if clause, like you have done in line 81 of this file. Do you agree?